Scripts/Python/get_device_list.py

```diff
@@ -34,61 +34,143 @@
    python get_device_list.py --ip <xx> --user <username> --password <pwd>
 """
 
-import sys
 import argparse
 from argparse import RawTextHelpFormatter
 import json
+import csv
+import os
 import requests
 import urllib3
-from requests.packages.urllib3.exceptions import InsecureRequestWarning
-requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
+from urllib3.exceptions import InsecureRequestWarning
+urllib3.disable_warnings(InsecureRequestWarning)
 
 
-def get_device_list(ip_address, user_name, password):
+class GetDeviceList:
     """ Authenticate with OME and enumerate devices """
-    try:
-        session_url = 'https://%s/api/SessionService/Sessions' % (ip_address)
-        base_uri = 'https://%s' %(ip_address)
-        device_url = base_uri + '/api/DeviceService/Devices'
-        next_link_url = None
-        headers = {'content-type': 'application/json'}
-        user_details = {'UserName': user_name,
-                        'Password': password,
+    def __init__(self, session_input, output_details):
+        self.__session_input = session_input
+        self.__output_details = output_details
+        if not self.__validateargs():
+            return
+
+        self.__base_uri = 'https://%s' % self.__session_input["ip"]
+        self.__headers = {'content-type': 'application/json'}
+
+        try:
+            self.__authenticate_with_ome()
+        except requests.exceptions.RequestException as auth_ex:
```

Comment:
When a session isn't established owing to incorrect credentials is an exception thrown? If not then would the script continue and attempt to get a list of devices even though a session wasn't created?

Reply:
Thanks. This is fixed.
- __authenticate_with_ome would return a boolean indicating success/Failure in obtaining a session.